File: 001/fizzBuzz.js
/*
	@author Sarah Kidd (www.linkedin.com/in/sarahkidd/)
	Description of problem at: https://www.codeeval.com/open_challenges/1/
*/

var fs = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line !== "") {
        line = line.replace(/\r/gm, ""); // For Windows.
        var arr = line.split(" "); // Splits the string by space
        var fizz = arr[0];
        var buzz = arr[1];
        var end = arr[2];

        for (var start = 1; start <= end; start++) {
            if (start % fizz === 0 && start % buzz === 0) {
                process.stdout.write("FB ");
            }
            else if (start % fizz === 0) {
                process.stdout.write("F ");
            }
            else if (start % buzz === 0) {
                process.stdout.write("B ");
            }
            else {
                process.stdout.write(start + " ");
            }
        }
        console.log();
    }
});

File: 001/fizzBuzz.cs
/*
	@author Sarah Kidd (www.linkedin.com/in/sarahkidd/)
	Description of problem at: https://www.codeeval.com/open_challenges/1/
*/

using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sarahTesting
{
    class Program
    {
        static void Main(string[] args)
        {
            using (StreamReader reader = File.OpenText(args[0]))
                while (!reader.EndOfStream)
                {
                    string line = reader.ReadLine();
                    if (null == line)
                        continue;
                    string[] arr = line.Split(' ');
                    int fizz = Convert.ToInt32(arr[0]);
                    int buzz = Convert.ToInt32(arr[1]);
                    int end = Convert.ToInt32(arr[2]);

                    for (int start = 1; start <= end; start++)
                    {
                        if (start % fizz == 0 && start % buzz == 0)
                            Console.Write("FB ");
                        else if (start % fizz == 0)
                            Console.Write("F ");
                        else if (start % buzz == 0)
                            Console.Write("B ");
                        else
                            Console.Write(start + " ");
                    }
                    Console.WriteLine();
                }
        }
    }
}
